python/lsst/daf/butler/core/_butlerUri/_butlerUri.py

```diff
@@ -674,30 +710,84 @@ def size(self) -> int:
         raise NotImplementedError()
 
     def __str__(self) -> str:
+        """Convert the URI to its native string form."""
         return self.geturl()
 
     def __repr__(self) -> str:
+        """Return string representation suitable for evaluation."""
         return f'ButlerURI("{self.geturl()}")'
 
     def __eq__(self, other: Any) -> bool:
+        """Compare supplied object with this `ButlerURI`."""
         if not isinstance(other, ButlerURI):
-            return False
+            return NotImplemented
         return self.geturl() == other.geturl()
 
+    def __hash__(self) -> int:
+        """Return hash of this object."""
+        return hash(str(self))
+
     def __copy__(self) -> ButlerURI:
+        """Copy constructor.
+
+        Object is immutable so copy can return itself.
+        """
         # Implement here because the __new__ method confuses things
-        # Be careful not to convert a relative schemeless URI to absolute
-        return type(self)(str(self), forceAbsolute=self.isabs())
+        return self
 
     def __deepcopy__(self, memo: Any) -> ButlerURI:
+        """Deepcopy the object.
+
+        Object is immutable so copy can return itself.
+        """
         # Implement here because the __new__ method confuses things
         return self.__copy__()
```

Comment:
???  Mismatch with docstring.

Reply:
Technically the docstring is correct because copy does return self... but given that you called me out on it I will fix it.